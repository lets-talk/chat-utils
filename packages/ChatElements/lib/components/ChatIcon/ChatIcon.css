@import '../../common/styles/colors.css';
@import '../../common/styles/variables.css';
@import './config.css';

.letstalk-ChatIconContainer {
  position: relative;
}

.letstalk-ChatIcon {
  position: relative;
  float: right;
  margin: 0 10px 50px 0;
  color: var(--color-primary);
  background-color: var(--chatIcon-background);
  border-radius: 50%;
  box-shadow: 0px 20px 13px -15px rgba(0,0,0,0.44);

  @apply --reset;
}

.letstalk-ChatIcon span.intro-message {
  font-size: var(--chatIcon-font-size);
  float: left;
  text-align: left;
  color: var(--chatIcon-color);
  width: calc(100% - 50px);
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.letstalk-ChatIcon i.md-icon-chat {
  float: right;
}

.letstalk-ChatIcon.default{
  border-radius: 0;
}

.letstalk-ChatIcon:focus {
  outline: none;
}
