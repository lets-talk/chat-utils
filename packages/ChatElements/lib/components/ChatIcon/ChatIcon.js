import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import IconButton from 'material-ui/IconButton';

import darkBaseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import styles from './ChatIcon.css';

// Stylesheets
import styled from 'styled-components';

const ChatIcon = (props) => {
  // Show the toggle icon only when widget is hidden or minimized and
  // animation is finished.
  let toggleIconDisplay = 'block';
  let toggleIconVisibility = 'hidden';

  if (props.display === 'minimized' || props.display === 'hidden') {
    toggleIconDisplay = 'block';
  }

  if (props.display === 'minimized' && props.animationStatus) {
    toggleIconVisibility = 'visible';
  }

  if (props.display === 'hidden') {
    toggleIconVisibility = 'visible';
  }

  if (props.display === 'small') {
    toggleIconDisplay = 'none';
  }

  const {
    imageUrl,
  } = props.chat_icon_pic;

  const {
    chatIconRadius = '0px',
  } = props;

  const useImage = !!imageUrl;

  const msgLbl = props.newMessages > 20 ? '20+' : props.newMessages;

  const messagesBubble = (
    <div
      style={props.bubbleMessageStyle}
    >{ msgLbl }
    </div>
  );

  const Button = styled.button`
    position: relative;
    margin-bottom: 50px;
    height: 50px;
    width: 50px;
    float: right;
    margin-right: 10px;
    color: ${(cprops) => cprops.theme.palette.primary1Color};
    background-color: ${(cprops) => cprops.theme.palette.primary1Color};
    border-radius: 50%;
    box-shadow: 0 3px 10px rgba(0, 0, 0, 0.16), 0 3px 10px rgba(0, 0, 0, 0.23);

    &:active {
      border-style: none;
    }

    &:focus {
      outline: none;
    }
`;

  const defaultIcon = (
    <div className={styles.ChatIconContainer}>
      <Button
        onTouchTap={(a) => a}
      >
        <i className="md-icon-chat" style={{ backgroundColor: 'transparent' }}></i>
      </Button>
      { props.newMessages > 0 && messagesBubble }
    </div>
  );

  const defaultIcon2 = (
    <MuiThemeProvider muiTheme={getMuiTheme(darkBaseTheme)}>
      <div>
        <IconButton
          className="chat-icon"
          style={{
            position: 'relative',
            marginBottom: '50px',
            height: '50px',
            float: 'right',
            marginRight: '10px',
            backgroundColor: props.theme.palette.primary1Color,
            borderRadius: '50%',
            boxShadow: '0px 20px 13px -15px rgba(0,0,0,0.44)',
          }}
        >
          <i className="md-icon-chat" style={{ backgroundColor: 'transparent' }}></i>
        </IconButton>
        { props.newMessages > 0 && messagesBubble }
      </div>
    </MuiThemeProvider>
  );

  const imageIcon = (
    <div
      className="chat-icon"
      style={{
        backgroundColor: 'transparent',
        height: '50px',
        width: 'auto',
        position: 'relative',
        marginBottom: '50px',
        float: 'right',
        marginRight: '10px',
        display: toggleIconDisplay,
        visibility: toggleIconVisibility,
        borderRadius: chatIconRadius,
        cursor: 'pointer',
        boxShadow: '0px 25px 13px -22px rgba(0,0,0,0.44)',
      }}
      iconStyle={{
        borderRadius: chatIconRadius, backgroundColor: 'transparent', height: '50px', width: 'auto',
      }}
      onTouchTap={(a) => a}
    >
      <img
        alt="Launch chat"
        src={imageUrl}
        style={{
          borderRadius: chatIconRadius, backgroundColor: 'transparent', height: '50px', width: 'auto',
        }}
      />
      { props.newMessages > 0 && messagesBubble }
    </div>
  );

  return (
    <div>
      { useImage && imageIcon }
      { !useImage && defaultIcon }
    </div>
  );
};


ChatIcon.propTypes = {
  /**
   * Description of prop "theme".
   */
  theme: PropTypes.shape({
    palette: PropTypes.shape({
      primary1Color: PropTypes.string,
      primary2Color: PropTypes.string,
      counterText: PropTypes.string,
    }),
  }),
  bubbleMessageStyle: PropTypes.object,
  chat_icon_pic: PropTypes.string,
  chatIconRadius: PropTypes.string,
  display: PropTypes.string,
  animationStatus: PropTypes.bool,
  newMessages: PropTypes.number,
};

ChatIcon.defaultProps = {
  theme: {
    palette: {
      primary1Color: '#f33535',
      cntColor: 'white',
      counterText: '',
    },
  },
  bubbleMessageStyle: {
    position: 'absolute',
    top: '0px',
    right: '0px',
    backgroundColor: '#f33535',
    color: 'white',
    borderRadius: '20px',
    padding: '1px',
    minWidth: '24px',
    textAlign: 'center',
    fontSize: '12px',
    fontFamily: 'sans-serif',
    lineHeight: '24px',
    fontWeight: '600',
  },
  chat_icon_pic: '',
  display: 'minimized',
};

export default ChatIcon;
