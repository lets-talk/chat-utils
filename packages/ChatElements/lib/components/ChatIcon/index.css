@import './config.css';

.letstalk-ChatIconContainer {
  position: relative;
}

.letstalk-ChatIcon {
  position: relative;
  float: right;
  margin: 0 10px 50px 0;
  color: var(--color-primary);
  background-color: var(--chatIcon-background);
  border-radius: 50%;
  box-shadow: 0px 20px 13px -15px rgba(0,0,0,0.44);

  box-sizing: border-box;
  font-family: var(--preferred-font);
  -webkit-font-smoothing: antialiased;
  font-smoothing: antialiased;
  text-size-adjust: 100%;

  & *,
  & *::after,
  & *::before {
    box-sizing: border-box;
    -webkit-font-smoothing: antialiased;
    font-smoothing: antialiased;
    text-size-adjust: 100%;
    -webkit-touch-callout: none;
  }
}

.letstalk-ChatIcon span.intro-message {
  font-size: var(--chatIcon-font-size);
  float: left;
  text-align: left;
  color: var(--chatIcon-color);
  width: calc(100% - 50px);
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.letstalk-ChatIcon i.md-icon-chat {
  float: right;
}

.letstalk-ChatIcon.default{
  border-radius: 0;
}

.letstalk-ChatIcon:focus {
  outline: none;
}
