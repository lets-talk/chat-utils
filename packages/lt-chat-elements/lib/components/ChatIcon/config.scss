@import "../../common/styles/variables";
@import "../../common/styles/mixins";
@import "../../common/styles/themes/themify";

$chatIcon-font-size: $font-size-footnote;
