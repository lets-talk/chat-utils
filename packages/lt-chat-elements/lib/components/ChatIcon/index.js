import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import MdChat from 'react-icons/lib/md/chat';

import { ellipsis, themeColor, textColor } from '../../utils/style';

const SimpleButton = styled.button`
  color: ${(props) => textColor(props.theme, 'dark', 'primary')};
  font-size: ${(props) => props.theme.typography.classes.footnote.fontSize};
  font-weight: ${(props) => props.theme.typography.weights.fontWeightBold};
  line-height: ${(props) => props.theme.typography.classes.footnote.lineHeight};
  background-color: ${(props) => themeColor(props.theme, 'accent', 'base')};
`;

const ChatIcon = (props) => {
  const {
    imageUrl,
  } = props;

  const {
    chatIconRadius = '0px',
    onClick,
  } = props;

  const useImage = !!imageUrl;

  const msgLbl = props.newMessages > 20 ? '20+' : props.newMessages;

  const StyledBubbleDiv = styled.div`
    position: absolute;
    top: 0px;
    right: 0px;
    background-color: ${(innerProps) => themeColor(innerProps.theme, 'primary', 'dark')};
    color: white;
    border-radius: 50%;
    padding: 1px;
    min-width: 24px;
    text-align: center;
    font-size: 12px;
    font-family: sans-serif;
    line-height: 24px;
    font-weight: 600;
  `;
  StyledBubbleDiv.displayName = 'StyledBubbleDiv';

  const messagesBubble = (
    <StyledBubbleDiv>{ msgLbl }</StyledBubbleDiv>
  );

  const StyledButton = SimpleButton.extend`
    width: ${props.width}px;
    height: ${props.height}px;
    margin: ${props.margin};
    border-radius: calc(${props.height}px / 2);
    line-height: calc(${props.height}px / 2);

    border-radius: ${() =>
    (props.type && props.type === 'rounded' && '5px 5px 0 0')
    || (props.type && props.type === 'circle' && '50%')
    || '0'};

    &:active {
      border-style: none;
    }

    &:focus {
      outline: none;
    }

    span {
      width: calc(100% - ${props.height}px);
      font-size: $chatIcon-font-size;
      float: left;
      text-align: left;
      ${ellipsis('100%')}
    }
`;
  StyledButton.displayName = 'StyledButton';

  const StyledImageIconContainer = styled.div`
    width: ${props.width}px;
    height: ${props.height}px;
    margin: ${props.margin};

    background-color: 'transparent';
    position: relative;
    margin-bottom: 50px;
    float: right;
    margin-right: 10px;
    borderRadius: ${chatIconRadius};
    cursor: pointer;
    boxShadow: 0px 25px 13px -22px rgba(0,0,0,0.44);

    img {
      width: 100%;
      height: 100%;
      border-radius: calc(${props.height}px / 2);
      line-height: calc(${props.height}px / 2);

      border-radius: ${() =>
    (props.type && props.type === 'rounded' && '5px 5px 0 0')
      || (props.type && props.type === 'circle' && '50%')
      || '0'};
    }
  `;
  StyledImageIconContainer.displayName = 'StyledImageIconContainer';

  const defaultIcon = (
    <div className="letstalk-ChatIconContainer">
      <StyledButton
        width={props.width}
        height={props.height}
        margin={props.margin}
        type={props.type}
      >
        <span>{props.text}</span>
        {props.showIcon && <MdChat size={props.height / 2.5} color="white" />}
      </StyledButton>
      { props.newMessages > 0 && messagesBubble }
    </div>
  );

  const imageIcon = (
    <StyledImageIconContainer>
      <img
        alt="Launch chat"
        src={imageUrl}
      />
      { props.newMessages > 0 && messagesBubble }
    </StyledImageIconContainer>
  );

  return (
    <div
      tabIndex="0"
      role="button"
      onKeyUp={onClick}
      onClick={onClick}
    >
      { useImage && imageIcon }
      { !useImage && defaultIcon }
    </div>
  );
};


ChatIcon.propTypes = {
  /**
   * types: default, rounded, circle
   */
  type: PropTypes.string,
  /**
   * text: Text to show on the component
   */
  text: PropTypes.string,
  /**
   * showIcon: Wheter or not display the chat icon
   */
  showIcon: PropTypes.bool,
  /**
   * width: Set the width of the button
   */
  width: PropTypes.number,
  /**
   * height: Set the height of the button
   */
  height: PropTypes.number,
  /**
   * margin: Set the margin of the button
   */
  margin: PropTypes.string,
  /**
   * onCLick: Hanlder function when click Event happens
   */
  onClick: PropTypes.func,
  /**
  * imageUrl: Image source of the image to show as chatIcon
  */
  imageUrl: PropTypes.string,
  /**
  * chatIconRadius: The border radius of the chat icon
  */
  chatIconRadius: PropTypes.string,
  /**
  * newMessages: Number of new messages to show
  */
  newMessages: PropTypes.number,
};

ChatIcon.defaultProps = {
  type: 'rounded',
  text: '',
  showIcon: true,
  onClick: null,
  height: 50,
  width: 250,
  margin: '',
  imageUrl: '',
};

export default ChatIcon;
