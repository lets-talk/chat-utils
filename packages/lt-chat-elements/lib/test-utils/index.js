// test-utils/index.js

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const defaultTheme = {
  palette: {
    common: {
      black: 'rgb(0, 0, 0)',
      white: 'rgb(255, 255, 255)',
      online: 'rgb(67, 160, 71)',
      offline: 'rgb(229, 57, 53)',
      sleeping: 'rgb(253, 216, 53)',
    },
    colors: {
      primary: {
        base: 'rgb(4, 38, 73)',
        light: 'rgb(86, 165, 246)',
        dark: 'rgb(0, 0, 0)',
      },
      accent: {
        base: 'rgb(255, 87, 50)',
        light: 'rgb(255, 171, 152)',
        dark: 'rgb(203, 37, 0)',
      },
      foreground: {
        base: 'rgb(94, 124, 139)',
        light: 'rgb(151, 173, 184)',
        dark: 'rgb(53, 70, 78)',
      },
      background: {
        base: 'rgb(255, 255, 255)',
        light: 'rgb(255, 255, 255)',
        dark: 'rgb(204, 204, 204)',
      },
    },
    contrastColors: {
      dark: {
        primary: 'rgb(255, 255, 255)',
        secondary: 'rgba(255, 255, 255, 0.7)',
        disabled: 'rgba(255, 255, 255, 0.5)',
        hint: 'rgba(255, 255, 255, 0.12)',
      },
      light: {
        primary: 'rgba(0, 0, 0, 0.87)',
        secondary: 'rgba(0, 0, 0, 0.54)',
        disabled: 'rgba(0, 0, 0, 0.38)',
        hint: 'rgba(0, 0, 0, 0.12)',
      },
    },
    error: {
      light: '#e57373',
      main: '#f44336',
      dark: '#d32f2f',
      contrastText: '#fff',
    },
    divider: 'rgba(255, 255, 255, 0.1)',
    action: {
      active: 'rgba(255, 255, 255, 0.9)',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.14)',
      disabled: 'rgba(255, 255, 255, 0.26)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)',
    },
  },
  shadows: {
    '2p': '0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 3px 1px -2px rgba(0, 0, 0, 0.2), 0 1px 5px 0 rgba(0, 0, 0, 0.12)',
    '3p': '0 3px 4px 0 rgba(0, 0, 0, 0.14), 0 3px 3px -2px rgba(0, 0, 0, 0.2), 0 1px 8px 0 rgba(0, 0, 0, 0.12)',
    '4p': '0 4px 5px 0 rgba(0, 0, 0, 0.14), 0 1px 10px 0 rgba(0, 0, 0, 0.12), 0 2px 4px -1px rgba(0, 0, 0, 0.2)',
    '6p': '0 6px 10px 0 rgba(0, 0, 0, 0.14), 0 1px 18px 0 rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.2)',
    '8p': '0 8px 10px 1px rgba(0, 0, 0, 0.14), 0 3px 14px 2px rgba(0, 0, 0, 0.12), 0 5px 5px -3px rgba(0, 0, 0, 0.2)',
    '16p': '0 16px 24px 2px rgba(0, 0, 0, 0.14), 0 6px 30px 5px rgba(0, 0, 0, 0.12), 0 8px 10px -5px rgba(0, 0, 0, 0.2)',
  },
  typography: {
    fontFamily: "'Nimbus sans-serif', 'Sans', 'Helvetica', 'Arial'",
    fontSize: 'calc(1 * 1px)',
    weights: {
      fontWeightUltraThin: 100,
      fontWeightThin: 300,
      fontWeightNormal: 300,
      fontWeightMedium: 500,
      fontWeightSemibold: 600,
      fontWeightBold: 700,
    },
    classes: {
      body: {
        fontSize: 'calc(16 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      caption: {
        fontSize: 'calc(12 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      footnote: {
        fontSize: 'calc(14 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      small: {
        fontSize: 'calc(10 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      headline: {
        fontSize: 'calc(24 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      subhead: {
        fontSize: 'calc(18 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      title: {
        fontSize: 'calc(20 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      jumbo: {
        fontSize: 'calc(44 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
      mega: {
        fontSize: 'calc(56 * 1px)',
        lineHeight: 'calc(16 * 1px)',
      },
    },
  },
  components: {
    conversationBox: {
      externalBackgroundColor: 'rgba(94, 124, 139, 0.1)',
      internalBackgroundColor: 'rgba(255, 213, 0, 0.2)',
      importantBackgroundColor: 'rgb(255, 246, 244)',
    },
    loader: {
      backgroundColor: 'rgb(252, 88, 48)',
    },
  },
};

const shallowWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();
  return shallow(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const mountWithTheme = (component) => {
  const context = shallow(<ThemeProvider theme={defaultTheme} />)
    .instance()
    .getChildContext();

  return mount(component, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  });
};

const renderWithTheme = (component) => renderer.create(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

const staticWithTheme = (component) => render(<ThemeProvider theme={defaultTheme}>{component}</ThemeProvider>);

export {
  defaultTheme,
  shallowWithTheme,
  mountWithTheme,
  staticWithTheme,
  renderWithTheme,
};
