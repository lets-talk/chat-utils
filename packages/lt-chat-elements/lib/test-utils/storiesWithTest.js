import { withTests } from '@storybook/addon-jest';
import jestTestResults from '../../tests-results/jest-test-results.json';

export default withTests({ results: jestTestResults }, {
  filesExt: '__tests__/**.js',
});
