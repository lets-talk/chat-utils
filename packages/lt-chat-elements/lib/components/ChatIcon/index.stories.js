import React from 'react';
// Storybook stuff
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, number, select, object } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';

import { WrapWithTheme, defaultTheme, themeOptions } from '../../utils/stories';
import withTests from '../../test-utils/storiesWithTest';
// Component to show on storybook
import { ChatIcon } from '../../../lib';

const stories = storiesOf('ChatIcon', module);
stories.addDecorator(withKnobs);
stories.addDecorator(withTests('ChatIcon'));

const defaultChatIconPic = {
  imageUrl: '',
};

stories.addWithInfo(
  'ChatIcon',
  'This is the basic usage with the ChatIcon with providing a label to show the text.',
  () => (
    <div style={{ position: 'fixed', bottom: 0, right: 0 }}>
      <WrapWithTheme
        themeName={select('Theme', themeOptions, defaultTheme)}
      >
        <ChatIcon
          type={select('Type', { default: 'default', rounded: 'rounded', circle: 'circle' }, 'rounded')}
          text={text('Text', 'En que podemos ayudar')}
          width={number('Width', 350)}
          height={number('Height', 50)}
          margin={text('Margin', '0px 10px 0px 0px')}
          onClick={action('onClick')}
          showIcon={boolean('Show Icon', true)}
          disabled={boolean('Disabled', false)}
          animationStatus={false}
          newMessages={number('New Messages', 22)}
          chat_icon_pic={object('Chat_Icon_Pic', defaultChatIconPic)}
        />
      </WrapWithTheme>
    </div>
  )
);
