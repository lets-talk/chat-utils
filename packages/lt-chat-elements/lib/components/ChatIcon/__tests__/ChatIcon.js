import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ChatIcon from '../index';
import { mountWithTheme } from '../../../test-utils';

describe('ChatIcon component', () => {
  it('should render with no props', () => {
    const component = shallow(<ChatIcon />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const props = {
      type: 'circle',
      text: '',
      showIcon: true,
      height: 50,
      width: 50,
      margin: '0px 10px 0px 0px',
      chat_icon_pic: '',
      display: 'minimized',
    };

    const component = shallow(<ChatIcon {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render Icon component', () => {
    const props = {
      type: 'circle',
      text: '',
      showIcon: true,
      height: 50,
      width: 50,
      margin: '0px 10px 0px 0px',
      chat_icon_pic: {},
      display: 'minimized',
    };

    const component = mountWithTheme(<ChatIcon {...props} />);

    expect(component.length).toBe(1);
    expect(component.find('StyledButton').length).toBe(1);
    expect(component.find('StyledImageIconContainer').length).toBe(0);
  });

  it('should render with Image', () => {
    const props = {
      type: 'circle',
      text: '',
      showIcon: true,
      height: 50,
      width: 50,
      margin: '0px 10px 0px 0px',
      imageUrl: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
      display: 'minimized',
    };

    const component = mountWithTheme(<ChatIcon {...props} />);

    expect(component.length).toBe(1);
    expect(component.find('StyledButton').length).toBe(0);
    expect(component.find('StyledImageIconContainer').length).toBe(1);
  });
});
