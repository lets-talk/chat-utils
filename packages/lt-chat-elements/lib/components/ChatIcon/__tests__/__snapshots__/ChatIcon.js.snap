// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatIcon component should render with custom props 1`] = `
<div
  onClick={null}
  onKeyUp={null}
  role="button"
  tabIndex="0"
>
  <div
    className="letstalk-ChatIconContainer"
  >
    <StyledButton
      height={50}
      margin="0px 10px 0px 0px"
      type="circle"
      width={50}
    >
      <span />
      <MdChat
        color="white"
        size={20}
      />
    </StyledButton>
  </div>
</div>
`;

exports[`ChatIcon component should render with no props 1`] = `
<div
  onClick={null}
  onKeyUp={null}
  role="button"
  tabIndex="0"
>
  <div
    className="letstalk-ChatIconContainer"
  >
    <StyledButton
      height={50}
      margin=""
      type="rounded"
      width={250}
    >
      <span />
      <MdChat
        color="white"
        size={20}
      />
    </StyledButton>
  </div>
</div>
`;
